dotsql.go

```diff
@@ -58,21 +59,35 @@ type ExecerContext interface {
 
 // DotSql represents a dotSQL queries holder.
 type DotSql struct {
-	queries map[string]string
+	queries map[string]*template.Template
+	data    any
 }
 
-func (d DotSql) lookupQuery(name string) (query string, err error) {
-	query, ok := d.queries[name]
+func (d DotSql) WithData(data any) DotSql {
+	return DotSql{queries: d.queries, data: data}
+}
+
+func (d DotSql) lookupQuery(name string, data any) (string, error) {
+	var query string
+	template, ok := d.queries[name]
 	if !ok {
-		err = fmt.Errorf("dotsql: '%s' could not be found", name)
+		return query, fmt.Errorf("dotsql: '%s' could not be found", name)
+	}
+	if template != nil {
```

Comment:
Not sure in which case `template == nil`, but still I think this function can be simplified like:
```go
	template, ok := d.queries[name]
	if !ok {
		return query, fmt.Errorf("dotsql: '%s' could not be found", name)
	}

	// XXX: fail here?
	if template == nil {
		return template, nil
	}

	buffer := bytes.NewBufferString("")
	if err := template.Execute(buffer, data); err != nil {
			return "", fmt.Errorf("parsing template: %w", err)
	}

	return buffer.String(), nil
```

also, note the error wrapping.

Reply:
The only case where it'll be nil is when there was an error parsing, but the switch makes sense generally.